sdk/core/core/inc/az_span.h

```diff
@@ -260,6 +260,10 @@ AZ_NODISCARD az_result az_span_to_uint64(az_span span, uint64_t* out_number);
  */
 AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number);
 
+AZ_NODISCARD az_result az_span_find(az_span span, az_span target, az_span* out_span);
+
+AZ_NODISCARD az_result az_span_token_next(az_span span, az_span delim, az_span* token, az_span* out_span);
```

Comment:
Agree for all comments, except about using uint8_t only for the delimiter. It would then need to be null terminated to search for it in the "source". 
Documentation is added now.

Reply:
I don't understand the null-terminated commented. Are you looking for a single character delimiter? You know the end due to the span's length.